msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -149,4 +188,22 @@
     return featureList;
 
   }
+
+  /** {@inheritDoc} */
+  @Override
+  public Float getFinishedPercentage() {
+    return objPeakDetection.getFinishedPercent();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<Feature> getResult() {
+    return finalFeatureList;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void cancel() {
+    objPeakDetection.cancel();
```

Comment:
You are right, I missed the null check afterwards.
However, it is much better to write
````
if (canceled) return null;
```` 
then
````
if (something == null) return null;
````
Because in the first case, anyone who is reading your code for the first time will understand that you are checking for the cancelation flag. In the second case, checking for null can mean any kind of error happened.